// msal/src/index.ts
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License.
 */

/**
 * @packageDocumentation
 * @module @azure/msal
 */

export { UserAgentApplication, authResponseCallback, errorReceivedCallback, tokenReceivedCallback } from "msal/src/UserAgentApplication";
export { Logger } from "msal/src/Logger";
export { LogLevel } from "msal/src/Logger";
export { Account } from "msal/src/Account";
export { Constants, ServerHashParamKeys } from "msal/src/utils/Constants";
export { Authority } from "msal/src/authority/Authority";
export { CacheResult } from "msal/src/UserAgentApplication";
export { CacheLocation, Configuration } from "msal/src/Configuration";
export { AuthenticationParameters } from "msal/src/AuthenticationParameters";
export { AuthResponse } from "msal/src/AuthResponse";
export { CryptoUtils } from "msal/src/utils/CryptoUtils";
export { UrlUtils } from "msal/src/utils/UrlUtils";
export { WindowUtils } from "msal/src/utils/WindowUtils";

// Errors
export { AuthError } from "msal/src/error/AuthError";
export { ClientAuthError } from "msal/src/error/ClientAuthError";
export { ServerError } from "msal/src/error/ServerError";
export { ClientConfigurationError } from "msal/src/error/ClientConfigurationError";
export { InteractionRequiredAuthError } from "msal/src/error/InteractionRequiredAuthError";
export { version } from "msal/src/packageMetadata";
